src/util/ieee_float.cpp

```diff
@@ -167,19 +167,25 @@ std::string ieee_floatt::format(const format_spect &format_spec) const
       mp_integer _exponent, _fraction;
       extract_base10(_fraction, _exponent);
 
-      if(_exponent>=0)
+      mp_integer adjusted_exponent=base10_digits(_fraction)+_exponent;
+
+      if(adjusted_exponent>=format_spec.precision ||
+         adjusted_exponent<-4)
```

Comment:
I don't actually know where that is from (was there before); and actual implementations appear to be different.